pkg/graveler/graveler.go

```diff
@@ -1659,6 +1661,31 @@ func (g *Graveler) Get(ctx context.Context, repository *RepositoryRecord, ref Re
 	for _, opt := range opts {
 		opt(&options)
 	}
+
+	if reference.CompactedBaseMetaRangeID != "" {
+		value, err := g.CommittedManager.Get(ctx, repository.StorageNamespace, reference.CompactedBaseMetaRangeID, key)
+		if err != nil {
+			return nil, err
+		}
+		// check if the key that we got from compaction is from compacted tokens or from committed
+		if options.StageOnly {
```

Comment:
Nice one!

Reply:
Also - notice that the value you found above, directly from staging, isn't necessarily from the staging area. It could of have the same value as the commit, consider changing the flow to perform the check for both options together..

Reply:
So it means we have a bug with the stage only flag?

Reply:
if we have, it means we also miss a test